dns/rdtypes/txtbase.py

```diff
@@ -76,6 +76,8 @@ def to_wire(self, file, compress=None, origin=None):
         for s in self.strings:
             l = len(s)
             assert l < 256
+            if not isinstance(s, bytes):
+                s = bytes(s, 'utf-8')
```

Comment:
How you can know that it is `utf-8` encoded?

Reply:
Ok I see, you are right! As far as I understood the loop underneath is catching all the chunks of a single TXT record (lower than 256 in length) and pass them to the write function. 

```
 70     def to_wire(self, file, compress=None, origin=None):
 71         for s in self.strings:
 72             l = len(s)
 73             assert l < 256
 74             file.write(struct.pack('!B', l))
 75             file.write(s)
```
With Python3 we face an issue when the TXT chunk is passed to file.write(s). It is expecting a byte-like object. 

```
  File "/home/lab/code/dnsmanager/env/lib/python3.5/site-packages/dns/rdtypes/txtbase.py", line 75, in to_wire
    file.write(s)
TypeError: a bytes-like object is required, not 'str'
```
How this can be resolved?

Reply:
What if file is opened in text mode instead of binary mode?